File: lab4/s1/web/src/main/java/ru/uvuv643/web/EnumValue.java
package ru.uvuv643.web;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;
import ru.uvuv643.web.EnumValueValidator;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Constraint(validatedBy = EnumValueValidator.class)
@Target({ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
public @interface EnumValue {
    Class<? extends Enum<?>> enumClass();
    String message() default "must be a valid enum (check DTO)";
    Class<?>[] groups() default {};
    Class<? extends Payload>[] payload() default {};
}

File: lab2/s1/src/main/kotlin/ru/uvuv643/soa/server/EnumValue.kt
import jakarta.validation.Constraint
import jakarta.validation.Payload
import ru.uvuv643.soa.server.EnumValueValidator
import kotlin.reflect.KClass

@Constraint(validatedBy = [EnumValueValidator::class])
@Target(AnnotationTarget.FIELD)
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
annotation class EnumValue(
    val enumClass: KClass<out Enum<*>>,
    val message:String = "must be valid enum (check DTO)",
    val groups:Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)